xrpl-std/src/core/types/amount/opaque_float.rs

```diff
@@ -0,0 +1,230 @@
+/// A struct that holds 8 opaque bytes supplied by `rippled`. This opaque value _actually_ holds an
+/// `exponent` and `mantissa` that rippled uses to represent a floating point number. Craft treats
+/// this as an opaque so that developers don't try and do arithmetic with these bytes directly, but
+/// instead use Host functions for any arithmetic on this type of number.
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+#[repr(C)]
+pub struct OpaqueFloat(pub [u8; 8]);
+
+impl OpaqueFloat {
+    // /// Accessor for the 8-bit `exponent` component of an `OpaqueFloat`.
```

Comment:
Note: I left this commented code in here to prod a discussion around whether we should allow craft to parse an `OpaqueFloat`.